arviz/data/inference_data.py

```diff
@@ -324,6 +378,322 @@ def sel(
+        for group in groups:
+            dataset = getattr(self, group)
+            valid_keys = set(name_dict.keys()).intersection(dataset.data_vars)
+            dataset = dataset.rename_vars({key: name_dict[key] for key in valid_keys})
+            setattr(out, group, dataset)
+        if inplace:
+            return None
+        else:
+            return out
+
+    def rename_dims(self, name_dict=None, groups=None, filter_groups=None, inplace=False):
+        """Perform xarray renaming of dimensions on all groups.
+
+        Loops groups to perform Dataset.rename_dims(name_dict)
+        for every key in name_dict if key is a dimension of the dataset.
+        The renaming is performed on all relevant groups (like
+        posterior, prior, sample stats) while non relevant groups like observed data are
+        omitted. See :meth:`xarray:xarray.Dataset.rename_dims`
+
+        Parameters
+        ----------
+        name_dict: dict
+            Dictionary whose keys are current dimension names and whose values are the desired
+            names.
+        groups: str or list of str, optional
+            Groups where the selection is to be applied. Can either be group names
+            or metagroup names.
+        filter_groups: {None, "like", "regex"}, optional, default=None
+            If `None` (default), interpret groups as the real group or metagroup names.
+            If "like", interpret groups as substrings of the real group or metagroup names.
+            If "regex", interpret groups as regular expressions on the real group or
+            metagroup names. A la `pandas.filter`.
+        inplace: bool, optional
+            If ``True``, modify the InferenceData object inplace,
+            otherwise, return the modified copy.
+
+
+        Returns
+        -------
+        InferenceData
+            A new InferenceData object with renamed dimension by default.
+            When `inplace==True` perform renaming in place and return `None`
+
+        """
+
+        groups = self._group_names(groups, filter_groups)
+        if "chain" in name_dict.keys() or "draw" in name_dict.keys():
+            raise KeyError("'chain' or 'draw' dimensions can't be renamed")
+
+        out = self if inplace else deepcopy(self)
+        for group in groups:
+            dataset = getattr(self, group)
+            valid_keys = set(name_dict.keys()).intersection(dataset.dims)
+            dataset = dataset.rename_dims({key: name_dict[key] for key in valid_keys})
+            setattr(out, group, dataset)
+        if inplace:
+            return None
+        else:
+            return out
+
+    # extending xarray.Dataset methods
+    set_index = extend_xr_method(xr.Dataset.set_index)
+    get_index = extend_xr_method(xr.Dataset.get_index)
+    reset_index = extend_xr_method(xr.Dataset.reset_index)
+    set_coords = extend_xr_method(xr.Dataset.set_coords)
+    reset_coords = extend_xr_method(xr.Dataset.reset_coords)
```

Comment:
can you add some more methods? The ones from the subset I am using the most or I think will be used the most but that are still missing are: `assign`, `assign_coords`, `sortby`

Some SO answers [1](https://stackoverflow.com/questions/61827061/xarray-make-two-dataarrays-in-the-same-dataset-use-the-same-coordinate-system), [2](https://stackoverflow.com/questions/59504320/how-do-i-subdivide-refine-a-dimension-in-an-xarray-dataset) of ArviZy calculations heavily requiring xarrray.